pkg/cmd/server/origin/master_config.go

```diff
@@ -128,6 +128,11 @@ type MasterConfig struct {
 // BuildMasterConfig builds and returns the OpenShift master configuration based on the
 // provided options
 func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
+	kubeEtcdClient, err := etcd.EtcdClient(*options.KubernetesMasterConfig.EtcdClientInfo)
```

Comment:
move this logic into `newServiceAccountTokenGetter`, where we do different things depending on whether `KubernetesMasterConfig` is present or not